anime_downloader/sites/helpers/selescrape.py

```diff
@@ -0,0 +1,193 @@
+        chrome_options.add_argument("--headless")
+        chrome_options.add_argument("--disable-gpu")
+        profile_path = os.path.join(data_dir, 'Selenium_chromium')
+        log_path = os.path.join(data_dir, 'chromedriver.log')
+        chrome_options.add_argument(f'--log-path {log_path}')
+        chrome_options.add_argument(f"--user-data-dir={profile_path}")
+        chrome_options.add_argument("--no-sandbox")
+        chrome_options.add_argument("--window-size=1920,1080")
+        chrome_options.add_argument(f'user-agent={get_random_header()}')
+        if binary == None:
+            if executable == None:
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(desired_capabilities=cap, options=chrome_options)
+        else:
+            if executable == None:
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(executable_path=binary, desired_capabilities=cap, options=chrome_options)
+    return driver
+
+
+def status_select(driver, url, status='hide'):
+    '''
+    For now it doesnt do what its name suggests, 
+    I have planned to add a status reporter of the http response code.
+    This part of the code is not removed because it is part of its core.
+    Treat it like it isnt here.
+    '''
+    try:
+        if status == 'hide':
+            driver.get(url)
+        elif status == 'show':
+            r = requests.head(url)
+            if r.status_code == 503:
+                raise RuntimeError("This website's sevice is unavailable or has cloudflare on.")
+            driver.get(url)
+            return r.status_code
+        else:
+            driver.get(url)
+    except requests.ConnectionError:
+        raise RuntimeError("Failed to establish a connection using the requests library.")
+
+        
+def cloudflare_wait(driver):
+    '''
+    It waits until cloudflare has gone away before doing any further actions.
+    The way it works is by getting the title of the page 
+    and as long as it is "Just a moment..." it will keep waiting.
+    This part of the code won't make the code execute slower 
+    if the target website has not a Cloudflare redirection.
+    At most it will sleep 1 second as a precaution. 
+    Also, i have made it time out after 30 seconds, useful if the target website is not responsive 
+    and to stop it from running infinitely.
+    '''
+    abort_after = 30 # The abort timer
+    start = time.time() # The time that the code starts to execute.
```

Comment:
Remove this useless comment.

Reply:
it is not useless. if anyone takes a look at the code and finds no comments then he will be like: what am i looking at?